### DISCUSSION TITLE
[데이터 공유] Elapsed time 결측치 채운 데이터

### DISCUSSION THREAD

**Discussion by ddoyoon** [bcaitech1/p4-dkt-lastffang#106] [Show and tell]
- 드라이브 링크
- 파일 설명
    -  `train_all_three.csv`, `test_all_three.csv`: 기존의 train_data.csv, test_data.csv 에 'elapsed' column 만 추가
    - `train_fe_all_three.csv`: 현재 master branch 기준 feature engineering 까지 적용
- 기타 
    - 추가한 'elapsed' column 을 사용하려면 `dkt/dataloader.py` 파일을 약간 수정해야 합니다. 수정본이 현재 PR #104 에 포함되어있는데, 혹시 merge 이전에 사용하신다면 해당 [PR의 `dkt/dataloader.py` 파일](https://github.com/bcaitech1/p4-dkt-lastffang/pull/104/files) 을 참고해서 수정 후 사용 부탁드립니다!
    - 제 서버, validation 기준 lstmattn 모델에서는 baseline 데이터에 비해 성능 향상이 약간 있었는데 다른 모델/서버/리더보드 상에서의 효과는 확실하지 않습니다 ㅠㅠ 혹시 사용해보신 분들은 baseline 데이터 대비 어떤 차이가 생겼는지 알려주시면 감사하겠습니다!
    - 아래는 stratified 5-fold 에서 baseline 데이터와 이 데이터로 수행한 결과를 비교한 그래프입니다.
![image](https://user-images.githubusercontent.com/18363932/121328931-b3bc6600-c94f-11eb-93b2-0ee040d59870.png)

**Comment by ddoyoon** [comment]
0초 및 10분 이상의 결측치들은 아래의 1~3 중 0초가 아닌 것들의 평균으로 대체했습니다.

유저 u 가 태그 t 인 문제 q 를 풀었을 때 elapsed 에 결측치가 있다면
1. 동일 문제 q 를 다른 유저들이 풀었을 때 걸린 평균 시간
2. 유저 u 가 태그 t 인 다른 문제들을 풀었을 때 걸린 평균 시간
3. 다른 유저들이 태그 t 인 다른 문제들을 풀었을 때 걸린 평균 시간

**Comment by newspring97** [comment]
lgbm에서는 elapsed를 적용하니 바로 성능이 떨어지네요ㅠㅠ

**Reply by ddoyoon** [reply]
엉엉 ㅠㅠ
